opentelemetry-sdk/tests/test_configurator.py

```diff
@@ -665,9 +677,151 @@ def test_logging_init_exporter(self):
         getLogger(__name__).error("hello")
         self.assertTrue(provider.processor.exporter.export_called)
 
+    @patch.dict(environ, {}, clear=True)
+    def test_otel_log_level_by_name_default(self):
+        self.assertEqual(_get_log_level(), NOTSET)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": "NOTSET "}, clear=True)
+    def test_otel_log_level_by_name_notset(self):
+        self.assertEqual(_get_log_level(), NOTSET)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " DeBug "}, clear=True)
+    def test_otel_log_level_by_name_debug(self):
+        self.assertEqual(_get_log_level(), DEBUG)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " info "}, clear=True)
+    def test_otel_log_level_by_name_info(self):
+        self.assertEqual(_get_log_level(), INFO)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " warn"}, clear=True)
+    def test_otel_log_level_by_name_warn(self):
+        self.assertEqual(_get_log_level(), WARNING)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " warnING "}, clear=True)
+    def test_otel_log_level_by_name_warning(self):
+        self.assertEqual(_get_log_level(), WARNING)
+
+    @patch.dict(environ, {"OTEL_LOG_LEVEL": " eRroR"}, clear=True)
+    def test_otel_log_level_by_name_error(self):
+        self.assertEqual(_get_log_level(), ERROR)
+
+
     @patch.dict(
         environ,
-        {"OTEL_RESOURCE_ATTRIBUTES": "service.name=otlp-service"},
+        {
+            "OTEL_RESOURCE_ATTRIBUTES": "service.name=otlp-service",
+            "OTEL_LOG_LEVEL": "CUSTOM_LOG_LEVEL",
```

Comment:
Does this line have an effect since the next line is patching `_get_log_level`?

Reply:
Yes because _get_log_level is only called if the env var is set.